server/test/repository/QuestionRepositoryTest.java

```diff
@@ -299,4 +300,201 @@ public void createOrUpdateDraft_managesUniversalTagCorrectly()
+            .build();
+
+    nameQuestion = repo.createOrUpdateDraft(nameQuestionDefinition);
+    dateQuestion = repo.createOrUpdateDraft(dateQuestionDefinition);
+    emailQuestion = repo.createOrUpdateDraft(emailQuestionDefinition);
+    phoneQuestion = repo.createOrUpdateDraft(phoneQuestionDefinition);
+
+    assertThat(
+            nameQuestion
+                .getQuestionTags()
+                .contains(PrimaryApplicantInfoTag.APPLICANT_NAME.getQuestionTag()))
+        .isTrue();
+    assertThat(
+            dateQuestion
+                .getQuestionTags()
+                .contains(PrimaryApplicantInfoTag.APPLICANT_DOB.getQuestionTag()))
+        .isTrue();
+    assertThat(
+            emailQuestion
+                .getQuestionTags()
+                .contains(PrimaryApplicantInfoTag.APPLICANT_EMAIL.getQuestionTag()))
+        .isTrue();
+    assertThat(
+            phoneQuestion
+                .getQuestionTags()
+                .contains(PrimaryApplicantInfoTag.APPLICANT_PHONE.getQuestionTag()))
+        .isTrue();
+
+    versionRepo.publishNewSynchronizedVersion();
+
+    nameQuestionDefinition =
+        new QuestionDefinitionBuilder(nameQuestion.getQuestionDefinition())
+            .setPrimaryApplicantInfoTags(ImmutableSet.of())
+            .build();
+    dateQuestionDefinition =
+        new QuestionDefinitionBuilder(dateQuestion.getQuestionDefinition())
+            .setPrimaryApplicantInfoTags(ImmutableSet.of())
+            .build();
+    emailQuestionDefinition =
+        new QuestionDefinitionBuilder(emailQuestion.getQuestionDefinition())
+            .setPrimaryApplicantInfoTags(ImmutableSet.of())
+            .build();
+    phoneQuestionDefinition =
+        new QuestionDefinitionBuilder(phoneQuestion.getQuestionDefinition())
+            .setPrimaryApplicantInfoTags(ImmutableSet.of())
+            .build();
+
+    nameQuestion = repo.createOrUpdateDraft(nameQuestionDefinition);
+    dateQuestion = repo.createOrUpdateDraft(dateQuestionDefinition);
+    emailQuestion = repo.createOrUpdateDraft(emailQuestionDefinition);
+    phoneQuestion = repo.createOrUpdateDraft(phoneQuestionDefinition);
+
+    assertThat(
+            nameQuestion
+                .getQuestionTags()
+                .contains(PrimaryApplicantInfoTag.APPLICANT_NAME.getQuestionTag()))
+        .isFalse();
+    assertThat(
+            dateQuestion
+                .getQuestionTags()
+                .contains(PrimaryApplicantInfoTag.APPLICANT_DOB.getQuestionTag()))
+        .isFalse();
+    assertThat(
+            emailQuestion
+                .getQuestionTags()
+                .contains(PrimaryApplicantInfoTag.APPLICANT_EMAIL.getQuestionTag()))
+        .isFalse();
+    assertThat(
+            phoneQuestion
+                .getQuestionTags()
+                .contains(PrimaryApplicantInfoTag.APPLICANT_PHONE.getQuestionTag()))
+        .isFalse();
+
+    // Update existing draft, ensure tags are correct
+    nameQuestionDefinition =
```

Comment:
suggestion (non-blocking): I think this is identical to the first chunk of tests so you could factor it out into a helper method.